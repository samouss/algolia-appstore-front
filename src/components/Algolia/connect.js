import React from 'react';
import PropTypes from 'prop-types';
import { getDisplayName } from 'core/utils';
import { ContextTypes } from './Provider';

export const ConnectPropTypes = {
  helper: PropTypes.object.isRequired,
};

export const connect = WrappedComponent => {
  const Connect = (props, context) => (
    <WrappedComponent
      {...props}
      helper={context.algoliaHelper}
    />
  );

  Connect.displayName = getDisplayName(
    WrappedComponent,
    'connect',
  );

  Connect.contextTypes = ContextTypes;

  return Connect;
};

export default connect;
