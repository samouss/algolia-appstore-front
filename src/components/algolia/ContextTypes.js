import PropTypes from 'prop-types';

export default {
  algoliaClient: PropTypes.object.isRequired,
  algoliaHelper: PropTypes.object.isRequired,
};
