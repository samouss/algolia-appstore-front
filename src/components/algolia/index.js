export { default as Provider } from './Provider';
export { default as withPaginateResults } from './withPaginateResults';
