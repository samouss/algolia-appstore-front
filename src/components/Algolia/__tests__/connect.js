import React from 'react';
import { shallow } from 'enzyme';
import { createMockAlgoliaClient, createMockAlgoliaHelper } from 'test/algolia';
import connect from '../connect';

describe('algolia', () => {
  describe('connect', () => {
    const createContext = () => ({
      algoliaHelper: createMockAlgoliaHelper(),
      algoliaClient: createMockAlgoliaClient(),
    });

    const Component = props => (
      <input {...props} />
    );

    it('expect to render', () => {
      const context = createContext();

      const props = {
        className: 'sample-class-name',
      };

      const ApplyComponent = connect(Component);

      const component = shallow(
        <ApplyComponent
          {...props}
        />,
        { context },
      );

      expect(component).toMatchSnapshot();
    });

    it('expect to have correct displayName', () => {
      const ApplyComponent = connect(Component);

      const expectation = 'connect(Component)';
      const actual = ApplyComponent.displayName;

      expect(actual).toBe(expectation);
    });
  });
});
