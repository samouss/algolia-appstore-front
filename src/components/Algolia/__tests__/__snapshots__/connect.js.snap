// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`algolia connect expect to render 1`] = `
<Component
  className="sample-class-name"
  helper={
    Object {
      "addDisjunctiveFacetRefinement": [Function],
      "addNumericRefinement": [Function],
      "clearRefinements": [Function],
      "getIndex": [Function],
      "getState": [Function],
      "nextPage": [Function],
      "on": [Function],
      "removeDisjunctiveFacetRefinement": [Function],
      "removeListener": [Function],
      "removeNumericRefinement": [Function],
      "search": [Function],
      "setIndex": [Function],
      "setQuery": [Function],
      "setQueryParameter": [Function],
    }
  }
/>
`;
